sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ResourceExtensions.cs

```diff
@@ -2,23 +2,27 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
 using System.Net;
+using Azure.Monitor.OpenTelemetry.Exporter.Models;
 using OpenTelemetry.Resources;
 
 namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
 {
     internal static class ResourceExtensions
     {
         private const string s_AiSdkPrefixKey = "ai.sdk.prefix";
+        private const int Version = 2;
 
-        internal static AzureMonitorResource? UpdateRoleNameAndInstance(this Resource resource)
+        internal static AzureMonitorResource? CreateAzureMonitorResource(this Resource resource, string? instrumentationKey = null)
```

Comment:
why ikey null here?
ikey is a required field of the connection string which is also required.

Reply:
`StandardMetricsExtractionProcessor` use `CreateAzureMonitorResource`. But `instrumentationKey` won't be available when we register `StandardMetricsExtractionProcessor`. Hence, kept here as optional.

Reply:
I see. thanks for the information.

Perhaps this can be a future enhancement. Our internals should be able to trust that an instrumentation key exists.